vertical-pod-autoscaler/pkg/admission-controller/certs.go

```diff
@@ -54,21 +62,40 @@ func (cr *certReloader) start(stop <-chan struct{}) error {
 		return err
 	}
 
-	if err = watcher.Add(path.Dir(cr.tlsCertPath)); err != nil {
+	if err = watcher.Add(cr.tlsCertPath); err != nil {
+		return err
+	}
+	if err = watcher.Add(cr.tlsKeyPath); err != nil {
 		return err
 	}
-	if err = watcher.Add(path.Dir(cr.tlsKeyPath)); err != nil {
+	if err = watcher.Add(cr.clientCaPath); err != nil {
 		return err
 	}
+
 	go func() {
 		defer watcher.Close()
 		for {
 			select {
 			case event := <-watcher.Events:
-				if event.Has(fsnotify.Create) || event.Has(fsnotify.Write) {
-					klog.V(2).InfoS("New certificate found, reloading")
-					if err := cr.load(); err != nil {
-						klog.ErrorS(err, "Failed to reload certificate")
+				// still support normal files to be written/created and reloaded
```

Comment:
I'm not sure I understand this comment

Reply:
The comment was supposed to say that we should still support regular files that are created/written (non-symlinked files). E.g., in the existing `certs_test.go`, the functionality is tested by manually creating the certificate files and checking if the reloader works, but in practice kubernetes does not mount the files like that.

If the comment is confusing or is too obvious, I can just remove it.

Reply:
I think the part that I didn't understand was the "still", it assumes that the reader knows the state of the code prior to the comment, and that's what git history is for